cmake/modules/dts.cmake

```
@@ -158,15 +169,14 @@ if(DTC_OVERLAY_FILE)
     )
 endif()
 
-set(i 0)
 foreach(dts_file ${dts_files})
-  if(i EQUAL 0)
-    message(STATUS "Found BOARD.dts: ${dts_file}")
-  else()
-    message(STATUS "Found devicetree overlay: ${dts_file}")
+  if(FOUND_BOARD_DTS AND ("${dts_file}" STREQUAL "${BOARD_DTS}"))
+    continue()                  # We already printed a message above.
   endif()
-
-  math(EXPR i "${i}+1")
+  if(HAD_DTS_SOURCE AND ("${dts_file}" STREQUAL "${ORIGINAL_DTS_SOURCE}"))
+    continue()
+  endif()
```

We know that above code would have printed any file present in `DTS_SOURCE` list.

So why not simply skip if the `dts_file` is part of that list, and avoid all the other book keeping.
```suggestion
  if(dts_file IN_LIST DTS_SOURCE)
    # Any file present in DTS_SOURCE has already be printed above
    continue()
  endif
```